packages/contracts-bedrock/scripts/L2Genesis2.s.sol

```diff
@@ -0,0 +1,583 @@
+        0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65, // 4
+        0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc, // 5
+        0x976EA74026E726554dB657fA54763abd0C3a0aa9, // 6
+        0x14dC79964da2C08b23698B3D3cc7Ca32193d9955, // 7
+        0x23618e81E3f5cdF7f54C3d65f7FBc0aBf5B21E8f, // 8
+        0xa0Ee7A142d267C1f36714E4a8F75612F20a79720, // 9
+        0xBcd4042DE499D14e55001CcbB24a551F3b954096, // 10
+        0x71bE63f3384f5fb98995898A86B02Fb2426c5788, // 11
+        0xFABB0ac9d68B0B445fB7357272Ff202C5651694a, // 12
+        0x1CBd3b2770909D4e10f157cABC84C7264073C9Ec, // 13
+        0xdF3e18d64BC6A983f673Ab319CCaE4f1a57C7097, // 14
+        0xcd3B766CCDd6AE721141F452C550Ca635964ce71, // 15
+        0x2546BcD3c84621e976D8185a91A922aE77ECEc30, // 16
+        0xbDA5747bFD65F08deb54cb465eB87D40e51B197E, // 17
+        0xdD2FD4581271e230360230F9337D5c0430Bf44C0, // 18
+        0x8626f6940E2eb28930eFb4CeF49B2d1F2C9C1199, // 19
+        0x09DB0a93B389bEF724429898f539AEB7ac2Dd55f, // 20
+        0x02484cb50AAC86Eae85610D6f4Bf026f30f6627D, // 21
+        0x08135Da0A343E492FA2d4282F2AE34c6c5CC1BbE, // 22
+        0x5E661B79FE2D3F6cE70F5AAC07d8Cd9abb2743F1, // 23
+        0x61097BA76cD906d2ba4FD106E757f7Eb455fc295, // 24
+        0xDf37F81dAAD2b0327A0A50003740e1C935C70913, // 25
+        0x553BC17A05702530097c3677091C5BB47a3a7931, // 26
+        0x87BdCE72c06C21cd96219BD8521bDF1F42C78b5e, // 27
+        0x40Fc963A729c542424cD800349a7E4Ecc4896624, // 28
+        0x9DCCe783B6464611f38631e6C851bf441907c710 // 29
+    ];
+
+    /// @notice Alias for `runWithStateDump` so that no `--sig` needs to be specified.
+    function run(Input memory _input) public {
+        address deployer = makeAddr("deployer");
+        vm.startPrank(deployer);
+        vm.chainId(_input.l2ChainID);
+
+        dealEthToPrecompiles();
+        setPredeployProxies(_input);
+        setPredeployImplementations(_input);
+        setPreinstalls();
+        if (_input.fundDevAccounts) {
+            fundDevAccounts();
+        }
+        vm.stopPrank();
+        vm.deal(deployer, 0);
+        vm.resetNonce(deployer);
+
+        Fork _fork = Fork(_input.fork);
+
+        if (!shouldApplyFork(_fork, Fork.DELTA)) {
+            return;
+        }
+
+        activateEcotone();
+
+        if (!shouldApplyFork(_fork, Fork.ECOTONE)) {
+            return;
+        }
+
+        activateFjord();
+
+        if (!shouldApplyFork(_fork, Fork.FJORD)) {
+            return;
+        }
+
+        if (!shouldApplyFork(_fork, Fork.GRANITE)) {
+            return;
+        }
+
+        if (!shouldApplyFork(_fork, Fork.HOLOCENE)) {
+            return;
+        }
+
+        activateIsthmus();
+
+        if (!shouldApplyFork(_fork, Fork.ISTHMUS)) {
+            return;
+        }
+
+        if (!shouldApplyFork(_fork, Fork.JOVIAN)) {
+            return;
+        }
+    }
+
+    function shouldApplyFork(Fork _latest, Fork _current) internal pure returns (bool) {
+        return _latest != _current;
```

Comment:
I saw the other comment saying that this approach minimizes the diff. I think this achieves that goal and is a bit semantically clearer (without the double negation in the logic).

```suggestion
        if (forkEquals(_fork, Fork.JOVIAN)) {
            return;
        }
    }

    function forkEquals(Fork _latest, Fork _current) internal pure returns (bool) {
        return _latest == _current;
```